src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -4901,6 +4902,14 @@ def close(self) -> None:
         except Exception as e:
             logger.warning(f"Task cleanup failed: {e}")
 
+        # ThreadPoolExecutor cleanup
+        try:
+            if hasattr(self, '_tool_executor') and self._tool_executor:
+                self._tool_executor.shutdown(wait=False, cancel_futures=True)
+                delattr(self, '_tool_executor')
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)